### PR TITLE
GFORMS-1954: Fix issue with 'Add or remove link' being overwritten with ordered list of ATL

### PR DESCRIPTION
**Pull request by lxol**

### CODE REVIEW

File: app/uk/gov/hmrc/gform/commons/MarkDownUtil.scala [hunk 1]
```diff
@@ -51,17 +52,31 @@ object MarkDownUtil {
     doc.body().html()
   }
 
-  def markDownParser(ls: LocalisedString)(implicit l: LangADT): Html = markDownParser(ls.value)
+  def parseMarkDownLocalisedString(ls: LocalisedString)(implicit l: LangADT): Html = parseMarkDownString(ls.value)
 
-  def markDownParser(ls: SmartString)(implicit sse: SmartStringEvaluator): Html = {
+  val noPreformatTagRenderer = new HtmlGenerator.DefaultTagRenderer(getDUMMY_ATTRIBUTES_CUSTOMIZER(), false) {
+    override def printHtml(html: CharSequence): CharSequence =
+      if (html == "<pre>" || html == "</pre>") "" else super.printHtml(html)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/commons/MarkDownUtil.scala:59]: Where is this `<pre>` tag coming from?

Comment (lxol) [reply app/uk/gov/hmrc/gform/commons/MarkDownUtil.scala:59]: > Where is this `\<pre\>` tag coming from?

It is generated by intellij markdown,  
https://github.com/JetBrains/markdown/blob/master/src/commonMain/kotlin/org/intellij/markdown/html/HtmlGenerator.kt
I created a custom TagRenderer that remove `pre` tag

Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/commons/MarkDownUtil.scala:59]: But what the source in our json template looked like?

Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/commons/MarkDownUtil.scala:59]: I don't see any `<pre>` being generated by that template? How exactly should I reproduce this?

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**